ui/lib/core/addon/components/icon.js

```diff
@@ -2,35 +2,54 @@
  * @module Icon
  * `Icon` components are glyphs used to indicate important information.
  *
+ * Flight icon documentation at https://flight-hashicorp.vercel.app/
+ *
  * @example
  * ```js
- * <Icon @glyph="cancel-square-outline" />
+ * <Icon @name="cancel-square-outline" />
  * ```
- * @param glyph=null {String} - The name of the SVG to render inline.
- * @param [size='m'] {String} - The size of the Icon, can be one of 's', 'm', 'l', 'xlm', 'xl', 'xxl'. The default is 'm'.
+ * @param name=null {String} - The name of the SVG to render inline.
+ * @param [sizeClass='m'] {String} - Used for sizing the hs-icon, can be one of 's', 'm', 'l', 'xlm', 'xl', 'xxl'. The default is 'm'.
+ * @param [size="16"] {String} - size for flight icon, can be 16 or 24
  *
  */
 import Component from '@ember/component';
 import { computed } from '@ember/object';
 import { assert } from '@ember/debug';
 import layout from '../templates/components/icon';
+import flightIconMap from '@hashicorp/flight-icons/catalog.json';
 
+const flightIconNames = flightIconMap.assets.mapBy('iconName').uniq();
 const SIZES = ['s', 'm', 'l', 'xlm', 'xl', 'xxl'];
 
 export default Component.extend({
   tagName: '',
   layout,
-  glyph: null,
-  size: 'm',
-  sizeClass: computed('size', function() {
-    let { size } = this;
+  name: null,
+  sizeClass: 'm', // hs-icon specific
+  size: null, // fight icon specific
```

Comment:
Is the default size inherited from Flight Icon? Should we be prescriptive about our default for Flight icons here?

Reply:
Yes the _FlightIcon_ default is 16. There is a `flightIconSize` computed property which gets passed to the component that will set the size to either 16 or 24 based on the `sizeClass` value if `size` was not provided which is the reason for the `null` default.

Reply:
👋  @zofskeez asked me to weigh in as well. If I understand right, I'm on board with our default being prescriptive to 16px; however, it seems like right now there are icons kicking around that are `small` (i.e. 10px) in seven or so instances. We shouldn't just change it and leave them there. What we could maybe do is split those out into a separate task to clean them up/go through them one by one and decide if a 16px icon or no icon is appropriate.

Reply:
@hashishaw the `sizeClass` property has been removed and `size` is now used exclusively with a default of 16 as of [this commit](https://github.com/hashicorp/vault/pull/12976/commits/5aa1cd3f294d748587cdda076e088ea1defbf100).